src/server/rdb_load.cc

```diff
@@ -2691,7 +2693,11 @@ error_code RdbLoader::LoadKeyValPair(int type, ObjSettings* settings) {
     item->is_sticky = settings->is_sticky;
     item->has_mc_flags = settings->has_mc_flags;
     item->mc_flags = settings->mc_flags;
-    item->expire_ms = settings->expiretime;
+    if (!ignore_expiry_) {
```

Comment:
I suggest doing it even earlier in the flow - line 2035 `settings.SetExpire(val);`